libs/contrib/Decidable/Order/Strict.idr

```diff
@@ -0,0 +1,84 @@
+||| An strict preorder (sometimes known as a quasi-order, or an
+||| ordering) is what you get when you remove the diagonal `{(a,b) | a
+||| r b , b r a}` from a preorder. For example a < b is an ordering.
+||| This module extends base's Decidable.Order with the strict versions.
+||| The interface system seems to struggle a bit with some of the constructions,
+||| so I hacked them a bit. Sorry.
+module Decidable.Order.Strict
+
```

Comment:
Not too fond of the `Hack` terminology. Is this not a free completion?

Reply:
What do you mean by `a free completion`?

Reply:
I mean that you are throwing in exactly the missing proofs to
get a partial order from a strict preorder.

I would call that [a completion](https://ncatlab.org/nlab/show/completion) of the strict preorder into a partial one.

And my hunch it's free comes from the "exactly the missing" and
not anything more.

Reply:
Not quite: each strict preorder has an associated poset via a coreflection (adding the missing diagonal) and each preorder has a strict preorder via a reflection (throwing away the diagonal). I think this adjunction restricts to an equivalence (actually isomorphism) when you restrict to posets only.

In any case the 'hack' isn't because this notion isn't canonical. It's because the type system can't seem to work out the correct instances and I need to use the 'hack' to construct them explicitly.

Reply:
OK, I've now removed the term 'hack' from the identifiers and called them `MkPreorder` and `InferPoset`, for example